Bio/SeqIO/GfaIO.py

```diff
@@ -29,7 +42,68 @@ def _check_tags(seq, tags):
                 )
 
 
-def GfaIterator(source):
+def _tags_to_annotations(tags):
+    """Build an annotations dictionary from a list of tags (PRIVATE)."""
+    annotations = {}
+    for tag in tags:
+        parts = tag.split(":")
+        if len(parts) < 3:
+            raise ValueError(f"Segment line has invalid tag: {tag}")
+        annotations[parts[0]] = parts[2]
+
```

Comment:
Add check that `parts[0]` does not contain any whitespace. Can happen if users add their own tags and delimit with spaces instead of tabs.
Would check whole tag for whitespace but I think JSON type tags can contain spaces.

Reply:
Good idea.
One thing in the spec that I just now noticed is that `parts[0]` should match the RegEx `[A-Za-z][A-Za-z0-9]`, which would cover the whitespace issue too.